pkg/api/mlflow/routes.go

```diff
@@ -57,25 +59,31 @@ const (
 
 // Router represents `mlflow` router.
 type Router struct {
-	prefixList []string
-	controller *controller.Controller
+	prefixList          []string
+	controller          *controller.Controller
+	namespaceRepository repositories.NamespaceRepositoryProvider
 }
 
 // NewRouter creates new instance of `mlflow` router.
-func NewRouter(controller *controller.Controller) *Router {
+func NewRouter(
+	controller *controller.Controller,
+	namespaceRepository repositories.NamespaceRepositoryProvider,
+) *Router {
 	return &Router{
 		prefixList: []string{
 			"/api/2.0/mlflow/",
 			"/ajax-api/2.0/mlflow/",
 		},
-		controller: controller,
+		controller:          controller,
+		namespaceRepository: namespaceRepository,
 	}
 }
 
 // Init makes initialization of all `mlflow` routes.
 func (r Router) Init(server fiber.Router) {
 	for _, prefix := range r.prefixList {
 		mainGroup := server.Group(prefix)
+		mainGroup.Use(namespaceMiddleware.New(r.namespaceRepository))
```

Comment:
moved middleware here, to connect it only to `mlflow` and `aim` routes instead of to all the possible like it was before.